lib/events.js

```diff
@@ -647,3 +649,79 @@ function once(emitter, name) {
     emitter.once(name, eventListener);
   });
 }
+
+const AsyncIteratorPrototype = ObjectGetPrototypeOf(
+  ObjectGetPrototypeOf(async function* () {}).prototype);
+
+function createIterResult(value, done) {
+  return { value, done };
+}
+
+function on(emitter, event) {
+  const unconsumedEvents = [];
+  const unconsumedPromises = [];
+  let error = null;
+
+  const iterator = ObjectSetPrototypeOf({
+    next() {
+      if (error) {
+        return Promise.reject(error);
```

Comment:
```suggestion
        return PromiseReject(error);
```

Reply:
Why?

Reply:
> Why?

So users can't mutate it, I think.
 https://github.com/nodejs/node/blob/eaf59754be421498287a7bf2471b83e629da1911/lib/internal/per_context/primordials.js#L5-L7

Reply:
Ah, just moving to primordials - got it